### PR TITLE
Add line length check

### PR DESCRIPTION
**Pull request by pirgeo**
07dcac8295eb613ac47bd33f62c45b35a2a9dee4 also contains updates to the GH action file, to use the latest versions of the checkout action. I also removed EOL'd node versions, and added node 18

### CODE REVIEW

**Comment by pirgeo** [comment]
Also updated the branch protection rules to only require node 14, 16, 18

**Review by joaopgrassi — APPROVED** [review]
I'm not that familiar with node versions so I'll leave that for others. The actual line length check looks good to me, so giving my approval on that.

**Review by pichlermarc — APPROVED** [review]

File: src/metric/base.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ limitations under the License.
 import { Dimension, Metric } from "./types";
 
 export abstract class BaseMetric<T> implements Metric {
+    private static metricLineMaxLength = 50_000;
```
Comment (dyladan) [inline src/metric/base.ts:20]: nit: static class members can't easily be minified. it is better to have a separate constant. Also uppercase names is the convention for constants so people know not to overwrite them.

```js
const METRIC_LINE_MAX_LENGTH = 50_000
```

**Review by dyladan — APPROVED** [review]

**Comment by dyladan** [comment]
Please update the version in package.json

**Comment by pirgeo** [comment]
I have changed it here: https://github.com/dynatrace-oss/dynatrace-metric-utils-js/pull/3/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R3
Did I miss another occurence?

**Review by dyladan — APPROVED** [review]

**Pull request merged**